File: inlay/android/src/main/kotlin/co/leancode/inlay/InlayPlugin.kt
package co.leancode.inlay

import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * No-op Flutter plugin class.
 *
 * This class exists solely so that Flutter's build system recognises
 * `inlay` as a plugin and bundles the Kotlin sources into
 * the AAR artifact. All actual engine/channel setup is performed
 * explicitly by [InlayNavigator].
 */
class InlayPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // No-op.
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // No-op.
    }
}

File: inlay/ios/Classes/InlayPlugin.swift
import Flutter

/// No-op Flutter plugin class.
///
/// This class exists solely so that Flutter's build system recognises
/// `inlay` as a plugin and bundles the Swift sources into
/// the iOS framework artifact. All actual engine/channel setup is
/// performed explicitly by `InlayNavigator`.
public class InlayPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        // No-op.
    }
}
